drake/systems/plants/@Manipulator/parameterEstimation.m

```diff
@@ -80,137 +124,140 @@
 qd=msspoly('qd',nq);
 qdd=msspoly('qdd',nq);
 u=msspoly('u',nu);
+t=msspoly('t',1);
 
 % Set up known parameters
 p=obj.getParamFrame.getPoly;
 pobj = setParams(obj,p);
 
 [H,C,B] = manipulatorDynamics(pobj,qt,qd);
-if strcmp(options.model,'dynamic')
-    % Formulate equations of motion
-    err = H*qdd + C - B*u;
-elseif strcmp(options.model,'energetic')
-    dt=msspoly('dt',1);
-    q1=msspoly('qo',nq);
-    q2=msspoly('qf',nq);
-    s1=msspoly('so',nq);
-    s2=msspoly('sf',nq);
-    c1=msspoly('co',nq);
-    c2=msspoly('cf',nq);
-    qt1=TrigPoly(q1,s1,c1);
-    qt2=TrigPoly(q2,s2,c2);
-    qd1=msspoly('qdto',nq);
-    qd2=msspoly('qdtf',nq);
-    % Formulate energy equations
-    [T1,U1] = energy(pobj,[qt1;qd1]);
-    [T2,U2] = energy(pobj,[qt2;qd2]);
-    % Need to formulate energy dissipation from AcrobotPlant class
-    dE = (B*u-[p(1);p(2)].*qd1)'*qd1*dt; % Under cursory testing, this works better
-%     dE = (B*u-[p(1);p(2)].*qd1)'*(q2-q1);
-    err = (T1+U1)-(T2+U2)+dE;
-else
-    error('Model not recognized')
-end
 
-[lp,M,Mb,lin_params,beta] = identifiableParameters(getmsspoly(err),p); % posynomial lumped params
-% [lp, M, Mb] = linearParameters(getmsspoly(err),p); % monomial lumped params
-
-nlp = length(lp);
-lp_orig = double(subs(lp,p,p_orig));
-lumped_params = msspoly('lp',nlp); 
-% now err=M*lp+Mb and lperr=M*lumped_params+Mb;
-
-
-%%   Step 2: Nonlinear least-squares estimation
-% 
-% Perform least squares on observable matrices: M_data*lp_est + Mb_data = 0
-
-if (nargin>1)
-  % populate A and b matrices from iddata
-  % todo: make the processing of q,qd,qdd more robust
-  Ts = get(data,'Ts');
-  t_data = get(data,'SamplingInstants')';
-  x_data = get(data,'OutputData')';
-  q_data = x_data(1:nq,:);
-  qd_data = x_data(nq+(1:nq),:);
-  if strcmp(options.model,'dynamic')
-    qdd_data = x_data(2*nq+(1:nq),:); 
-  end
-  u_data = get(data,'InputData')';
+if isDynamic || isEnergetic
+    if isDynamic
+        % Formulate equation error from equations of motion
+        err = H*qdd + C - B*u;
+    elseif isEnergetic
+        % If need memory efficiency, can cut down on # of msspoly
+        % this is just a bit more readable
+        % Initialize msspoly variables for expressing dE = E(t_2)-E(t_1)
+        dt=msspoly('dt',1);
+        
+        q1=msspoly('qo',nq);
+        s1=msspoly('so',nq);
+        c1=msspoly('co',nq);
+        qt1=TrigPoly(q1,s1,c1);
+        qd1=msspoly('qdto',nq);
+        
+        q2=msspoly('qf',nq);
+        s2=msspoly('sf',nq);
+        c2=msspoly('cf',nq);
+        qt2=TrigPoly(q2,s2,c2);
+        qd2=msspoly('qdtf',nq);
+        
+        % Formulate energy equations
+        [T1,U1] = energy(pobj,[qt1;qd1]);
+        [T2,U2] = energy(pobj,[qt2;qd2]);
+        
+        % ACROBOT-SPECIFIC FORMULATION - MUST CHANGE
```

Comment:
at very least you should throw a useful message if someone gets here (with a different plant)